fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/AgilityContext.java

```diff
@@ -194,9 +198,9 @@ class Agile implements AgilityContext {
         private long prevCommitCheckTime;
         private boolean closed = false;
 
-        protected Agile(FDBRecordContext callerContext, final long timeQuotaMillis, final long sizeQuotaBytes) {
+        protected Agile(FDBRecordContext callerContext, @Nullable FDBRecordContextConfig.Builder contextBuilder, final long timeQuotaMillis, final long sizeQuotaBytes) {
             this.callerContext = callerContext;
```

Comment:
Why are we still getting a `callerContext` if we are getting a `contextBuilder`?
It looks like the `callerContext` provides other things that you can't get from the `FDBRecordContextConfig`... Probably would be good to figure out how to clean that up so that this no longer references the `callerContext` after creation, that seems like a risk, but I don't think we want it in this PR.

Reply:
We need the `callerContext` for stats and `getCallerContext()`. The rational is that the agility context vanishes within the operation, but some operations need the one that is visible to the caller.